src/math.rs

```diff
@@ -428,3 +486,54 @@ mod tests {
         assert_eq!(arbitrary_obb.separating_axes.len(), 15);
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct Frustum<S: BaseFloat> {
+    matrix: Matrix4<S>,
+    frustum: collision::Frustum<S>,
+}
+
+impl<S: BaseFloat> Frustum<S> {
+    pub fn new(matrix: Matrix4<S>) -> Self {
+        Frustum {
+            matrix,
+            frustum: collision::Frustum::from_matrix4(matrix).unwrap(),
+        }
+    }
+}
+
+impl<S> PointCulling<S> for Frustum<S>
+where
+    S: BaseFloat + 'static + Sync + Send,
+{
+    fn contains(&self, point: &Point3<S>) -> bool {
+        contains(&self.matrix, point)
+    }
+
+    fn intersects(&self, aabb: &Aabb3<S>) -> bool {
+        match self.frustum.contains(aabb) {
+            Relation::Cross => true,
+            Relation::In => true,
+            Relation::Out => false,
+        }
+    }
+
+    fn transform(&self, isometry: &Isometry3<S>) -> Box<PointCulling<S>> {
+        let matrix = Matrix4::from(Decomposed {
```

Comment:
Here we could nicely make use of the conversion methods between `Decomposed` and `Isometry3`, which were deleted above...

Reply:
done